pkg/container-collection/options.go

```diff
@@ -432,3 +438,280 @@ func WithLinuxNamespaceEnrichment() ContainerCollectionOption {
+			}
+
+			// This function does not support array values
+			if len(values) != 1 {
+				return "", fmt.Errorf("unexpected number of matches for field %s: %d (expecting 1)",
+					field, len(values))
+			}
+
+			var strValues []string
+			for _, val := range values {
+				strVal, ok := val.(string)
+				if !ok {
+					log.Errorf("failed to convert to string field %s: %s", field, val)
+					continue
+				}
+				strValues = append(strValues, strVal)
+			}
+
+			return strValues[0], nil
+		}
+
+		getOwnerReference := func(resourceJSON interface{}) (*metav1.OwnerReference, error) {
+			values, err := getResourceField(".metadata.ownerReferences", resourceJSON)
+			if err != nil {
+				return nil, fmt.Errorf("failed to retrieve owner references: %w", err)
+			}
+
+			// No owner reference found
+			if len(values) == 0 {
+				return nil, nil
+			}
+
+			// TODO: How should we manage multiple references?
+			if len(values) != 1 {
+				log.Warnf("Taking only first owner reference and ignoring the rest: %v", values)
+			}
+
+			// Convert it into JSON to then use metav1.OwnerReference
+			var valueBytes []byte
+			valueBytes, err = json.Marshal(values[0])
+			if err != nil {
+				return nil, fmt.Errorf("failed to marshal owner references %v: %w", values[0], err)
+			}
+
+			ownerReferences := []metav1.OwnerReference{}
+			err = json.Unmarshal(valueBytes, &ownerReferences)
+			if err != nil {
+				return nil, fmt.Errorf("failed to unmarshal owner references %v: %w",
+					string(valueBytes), err)
+			}
+
+			return &ownerReferences[0], nil
+		}
+
+		// Future containers
+		cc.containerEnrichers = append(cc.containerEnrichers, func(containerDefinition *pb.ContainerDefinition) bool {
+			params := &ResourceParameters{
+				groupVersion: schema.GroupVersionResource{
+					Group:    "",
+					Version:  "v1",
+					Resource: "pods",
+				},
+				namespace: containerDefinition.Namespace,
+				listOpt: metav1.ListOptions{
+					FieldSelector: fields.OneTermEqualSelector("spec.nodeName", nodeName).String(),
+				},
+			}
+
+			pods, err := getResourceList(dynamicClient, params)
+			if err != nil {
+				log.Errorf("couldn't fetch pods: %s", err)
+				return true
+			}
+
+			// Fill owner references data
+			for _, pod := range pods {
+				podName, err := getStringField(".metadata.name", pod)
+				if err != nil {
+					log.Errorf("couldn't retrieve pod name: %s", err)
+					continue
+				}
+
+				uid, err := getStringField(".metadata.uid", pod)
+				if err != nil {
+					log.Errorf("couldn't retrieve uid of %q: %s", podName, err)
+					continue
+				}
+
+				// Check if this container is associated to this pod
+				uidWithUnderscores := strings.ReplaceAll(uid, "-", "_")
+				if !strings.Contains(containerDefinition.CgroupV2, uidWithUnderscores) &&
+					!strings.Contains(containerDefinition.CgroupV2, uid) &&
+					!strings.Contains(containerDefinition.CgroupV1, uidWithUnderscores) &&
+					!strings.Contains(containerDefinition.CgroupV1, uid) {
+					continue
+				}
```

Comment:
Why do you have to make this check? At this point the podname, namespace and other k8s fields should be set up, so you don't have to rely on cgroups anymore at this point.